components/Navbar/LoginButton.tsx

```diff
@@ -1,68 +1,212 @@
+import { useState, ReactNode, MouseEventHandler } from 'react'
 import Link from 'next/link'
 import clsx from 'clsx'
-import { RawButton } from 'components/Button'
 import { LoginContext } from 'contexts/LoginContext'
-// import { useState, useEffect } from 'react'
 
-export const LoginButton = () => {
-  /*
-  const [$loaded, $setLoaded] = useState<boolean>(false)
-  useEffect(() => {
-    const t = setTimeout(() => {
-      if (!$loaded) {
-        $setLoaded(true)
-      }
-    }, 3e3)
-    return () => clearTimeout(t)
-    })
-  */
+// TODO: disable this before merging!
+import useQuery from 'hooks/useQuery'
+
+import ChevronDown from './ChevronDown'
+import { ApiUserMetadata } from 'apiClient/types'
+import styles from './LoginButton.module.css'
+
+type ButtonProps = {
+  children: ReactNode
+  colorClassName?: string
+  className?: string
+  onClick?: MouseEventHandler<HTMLButtonElement>
+  onMouseOver?: MouseEventHandler<HTMLButtonElement>
+  onMouseOut?: MouseEventHandler<HTMLButtonElement>
+  border?: string
+}
+export const HoverButton = ({
+  children,
+  className = '',
+  border = `border-2`,
+  onMouseOver,
+  onMouseOut,
+}: ButtonProps) => {
   return (
-    <LoginContext.Consumer>
-      {({ status, checkLoggedIn, error, metadata }) => {
-        if (error) {
-          // TODO: Find a way to resolve this better, but keep for now
-          // eslint-disable-next-line no-console
-          console.log({ error, status })
-        }
+    <button
+      onMouseOver={onMouseOver}
+      onMouseOut={onMouseOut}
+      className={clsx(
+        styles.buttonRoot,
+        `flex`,
+        `justify-center`,
+        `items-center`,
+        `font-extended`,
+        `rounded-full`,
+        `whitespace-nowrap`,
+        `transition`,
+        `border-black`,
+        border,
+        className
+      )}
+    >
+      {children}
+    </button>
+  )
+}
+type StaticProps = {
+  href: string
+}
+
+const StaticButton = ({ href }: StaticProps) => (
+  <Link href={href} passHref>
+    <a>
+      <span>
+        Login<span className="md:hidden"> to Testnet</span>
+      </span>
+    </a>
+  </Link>
+)
+
+interface ApiUserMetadataUI extends ApiUserMetadata {
+  visible: boolean
+}
 
-        const loggedIn = checkLoggedIn()
+const HitState = () => <div className={styles.hitState} />
 
-        const goto =
-          loggedIn && metadata && metadata.id
-            ? `/users/${metadata.id}`
-            : '/login'
-        const hasGraffiti = metadata && !!metadata.graffiti
+const UserButton = ({ id, graffiti, visible }: ApiUserMetadataUI) => (
+  <div
+    className={clsx(
+      styles.userButton,
+      'absolute',
+      'flex',
+      'flex-col',
+      'top-[0.75rem]',
+      'pointer-events-none',
+      'md:pointer-events-auto',
```

Comment:
> Hovering over the Sign Out text in mobile/narrow view sets the username pill to the hovered state.

@dguenther So this one is a little tricky, basically I eat the hover events on mobile / base styling and then swap back to regular at the next breakpoint. However, on desktop at mobile breakpoints, because of the nature of `pointer-events` I wasn't able to make it so that the mouse looks right.

![pointerless](https://user-images.githubusercontent.com/18919/144687696-ec4fdee7-a16d-483b-99d0-1eabf50dc903.gif)
 
I think that's about the best I can do without investing a bunch more time and it's a smaller subset of the original problem b/c on mobile you won't have any hover events either way.

Reply:
Sounds good, that change seems like an improvement to me, I'm okay punting on this for now.